i3lock.c

```diff
@@ -1526,6 +1725,17 @@ bool load_slideshow_images(const char *path) {
     return true;
 }
 
+void gif_anim_loop(struct ev_loop *loop, struct ev_timer *timer, int delay) {
+    static int img_count = 0;
+
+    if (++img_count >= gif_img_count) img_count = 0;
+    img = gif_img[img_count].img;
+    redraw_screen();
```

Comment:
Will this cause issues on very-high-framerate GIFs?

Reply:
It might cause issues in that case, I believe. But the rate should be quite high and it depends on hardware. I thought that we can skip some frames if the rate greater than 1/60 but then I remembered that in GIF the delays between frames can vary and independent of each other. Not sure if there is an easy way to skip frames for that case..